src/scripts/destinyTrackerApi/reviewsFetcher.js

```diff
@@ -0,0 +1,98 @@
+import _ from 'underscore';
+import { itemTransformer } from './itemTransformer.js';
+
+class reviewsFetcher {
+  constructor($q, $http, trackerErrorHandler, loadingTracker, scoreMaintainer) {
+    this.$q = $q;
+    this.$http = $http;
+    this._itemTransformer = new itemTransformer();
+    this._trackerErrorHandler = trackerErrorHandler;
+    this._loadingTracker = loadingTracker;
+    this._reviewDataCache = scoreMaintainer;
+  }
+
+  getItemReviewsCall(item) {
+    return {
+      method: 'POST',
+      url: 'https://reviews-api.destinytracker.net/api/weaponChecker/reviews',
+      data: item,
+      dataType: 'json'
+    };
+  }
+
+  getItemReviewsPromise(item) {
+    var postWeapon = this._itemTransformer.getRollAndPerks(item);
+
+    var promise = this.$q
+              .when(this.getItemReviewsCall(postWeapon))
+              .then(this.$http)
+              .then(this._trackerErrorHandler.handleErrors, this._trackerErrorHandler.handleErrors)
+              .then((response) => { return response.data; });
+
+    this._loadingTracker.addPromise(promise);
+
+    return promise;
+  }
+
+  getUserReview(reviewData) {
+    return _.findWhere(reviewData.reviews, { isReviewer: true });
+  }
+
+  attachReviews(item,
+                reviewData) {
+    var userReview = this.getUserReview(reviewData);
+
+    item.communityReviews = reviewData.reviews;
+
+    if (userReview) {
+      item.userRating = userReview.rating;
+      item.userReview = userReview.review;
+      item.userReviewPros = userReview.pros;
+      item.userReviewCons = userReview.cons;
+    }
+
+    this._reviewDataCache.addReviewsData(item,
+                                         reviewData);
+  }
+
+  attachCachedReviews(item,
+                      cachedItem) {
+    item.communityReviews = cachedItem.reviews;
+
+    this.attachReviews(item,
+                       cachedItem);
+
+    if (cachedItem.userRating) {
+      item.userRating = cachedItem.userRating;
+    }
+
+    if (cachedItem.review) {
+      item.userReview = cachedItem.review;
+    }
+
+    if (cachedItem.pros) {
+      item.userReviewPros = cachedItem.pros;
+    }
+
+    if (cachedItem.cons) {
+      item.userReviewCons = cachedItem.cons;
+    }
```

Comment:
I guess it depends on whether you have those properties around (but with a null or undefined value) rather than having a sparse object you can merge in.